### PR TITLE
First-class Alias Support

### LINKED ISSUE #1: First-class command alias support
Currently, the way to do "aliases" is to just mount the same command module under a different name. Not nice.

In the `Nosedrum.Storage` behaviour:
- Add an optional callback to add & remove aliases, take aliases into account when looking up commands
- Add an optional callback used for querying for only commands / only aliases

In the `Nosedrum.Invoker.Split` implementation, make adjustments as necessary.

### PR DESCRIPTION
**Pull request by Bentheburrito**
Close #1 

At the moment, the `Storage` behaviour is untouched, which means users who implement their own `Storage` can choose whether they want to do anything with an `aliases` callback in their commands.

I know `if`s aren't the go-to for conditionals in Elixir, so if you know of a more functional approach (or feedback in general), please lmk ;D

Also, a bit unrelated, but about half of the unit tests aren't run with `mix test` because of errors like this:
```elixir
???????

  7) Nosedrum.Converters.ChannelTest: failure on setup_all callback, all tests have been invalidated
     ** (RuntimeError) failed to start child with the spec Nostrum.Cache.CacheSupervisor.
     Reason: already started: #PID<0.246.0>
     stacktrace:
       (ex_unit 1.12.1) lib/ex_unit/callbacks.ex:433: ExUnit.Callbacks.start_supervised!/2
       test/nosedrum/converters/channel_test.exs:12: Nosedrum.Converters.ChannelTest.__ex_unit_setup_all_0/1
       test/nosedrum/converters/channel_test.exs:1: Nosedrum.Converters.ChannelTest.__ex_unit__/2
```
I imagine Nostrum capped how many `Nostrum.Cache.CacheSupervisor`s can be spawned since the tests were written?

### CODE REVIEW

File: lib/nosedrum/command.ex [hunk 1]
```diff
@@ -129,5 +129,13 @@ defmodule Nosedrum.Command do
   """
   @callback command(msg :: Message.t(), args :: [String.t()] | any()) :: any()
 
-  @optional_callbacks [parse_args: 1]
+  @doc """
+  An optional callback that returns a list of aliases for the command.
```
Comment (jchristgit) [inline lib/nosedrum/command.ex:133]: Can we add a line here explaining that `Nosedrum.Storage` implementations use this in order to install a command at additional paths when `add_command/2` or `remove_command/2` is called?

Having similar documentation in `add_command/2` and `remove_command/2` would be great: I think just a sentence saying that "if a command has the c:Nosedrum.Command.aliases/0 callback defined, they will also be added / removed." would be sufficient.

**Review by jchristgit — CHANGES_REQUESTED** [review]
This is really really really cool, thank you so much!

I've taken this for a spin in bolt and realized meanwhile that I'm using my own library wrongly, and it just worked by accident previously. Fair enough!

As for your test failure - I ran local unit tests without a problem, but I've noticed that there's a flaky test because of the ETS table. I've changed the test case to not run async and all the tests are green now, including this branch.

Apart from the documentation per my comment, this is really nice!

I know they're a different horse to tackle, but what do you think on us handling aliases for command groups? I think it's a bit out of scope for this PR, but I'm curious on implementation thoughts. Maybe we could respect the `:default` command's `aliases`, but that would require you to implement one. Or we add an `:aliases` option, but that's kind of hacky...

File: lib/nosedrum/command.ex (hunk 1 above)
Comment (Bentheburrito) [reply lib/nosedrum/command.ex:133]: For sure! Just pushed a commit - lmk if those are good enough or you'd like some more detail to them (or resolve if they suite your fancy :slightly_smiling_face: )

**Comment by Bentheburrito** [comment]
I'm curious how you were using the lib wrong haha, and does it affect this PR?

Hmm the same tests are "invalidated" with your changes, but I guess it's (fortunately[?]) just an issue on my machine, so I'll look into it some more.

On command group aliases, what would be their use-case exactly? My understanding is that you use command_groups to organize commands (and maybe have many commands with the same name, but under different groups to avoid conflict). Though I only use top-level commands myself, so maybe I'm missing something important about nesting commands?

File: lib/nosedrum/command.ex (hunk 1 above)
Comment (jchristgit) [reply lib/nosedrum/command.ex:133]: This is excellent - thank you!

**Review by jchristgit — APPROVED** [review]
This looks good now. Thank you!

**Comment by jchristgit** [comment]
> I'm curious how you were using the lib wrong haha, and does it affect this PR?

Not directly - previously in bolt I was adding command groups via something like

```elixir
add_command(["group"], %{"subcommand" => SomeCommand})
```
That was kind of unsupported in the first place!

**Comment by jchristgit** [comment]
> On command group aliases, what would be their use-case exactly? My understanding is that you use command_groups to organize commands (and maybe have many commands with the same name, but under different groups to avoid conflict). Though I only use top-level commands myself, so maybe I'm missing something important about nesting commands?

I'll go with a practical example in bolt: There's a command group called `keeper` for the "gatekeeper" functionality. There's an alias called `gatekeeper` and another called `gk`, to make calling it simpler 🙂

**Comment by jchristgit** [comment]
Thank you so much!

**Comment by Bentheburrito** [comment]
> Not directly - previously in bolt I was adding command groups via something like
> 
> ```elixir
> add_command(["group"], %{"subcommand" => SomeCommand})
> ```
> 
> That was kind of unsupported in the first place!

Great example of "it's not a bug, it's a feature!" :P

> I'll go with a practical example in bolt: There's a command group called `keeper` for the "gatekeeper" functionality. There's an alias called `gatekeeper` and another called `gk`, to make calling it simpler slightly_smiling_face

Ahh I see. I also saw your messages in #elixir_nostrum about this, and I assume you might be going in the direction of nested lists like `[["gatekeeper", "gk"], "command_name"]`? If so, maybe we can drop the `aliases/0` callback altogether and include them in the command path:
```elixir
add_command([["gatekeeper", "gk", "keeper"], ["name", "alias1"]], CommandModule)
```
What do you think? Nested lists aren't the nicest thing to work with from a user's perspective imo, but I can't think of a way to avoid them with variable-depth trees like this

**Pull request merged**